### ISSUE TITLE
Cinder Rotation of Camera when WorldUp() changed

### ISSUE THREAD

**Issue by bradrothenberg** [cinder/Cinder#879]
Hi,

You could potentially argue that this is not an issue, but when you change the worldUp() of a mayaCam (now Camera), the axes for rotation do not switch to adjust for the new world up. It would seem that if you change the worldUp() of the camera, the axes should update so the mouse inputs seem correct to a user.

This also has to do with the best practice to switch between Y-up & Z-up...

Bradley

**Comment by paulhoux** [comment]
I'll take a look. Thanks for reporting, Bradley.

**Comment by paulhoux** [comment]
Hi Bradley,

so... when I call ```CameraPersp::setWorldUp()```, the orientation of the camera is changed accordingly. If I then call ```CameraPersp::getBillboardVectors()```, it correctly gives me the up and right vectors. So that all seems to work.

Could you show me in code what you're trying to do, what you'd expect and what you get instead?

**Comment by bradrothenberg** [comment]
Hi Paul,

The issue I was having was with how the CameraUI responds to mouse inputs once you change the SetWorldUp of the camera -- So if I set the SetWorldUp to {0,0,1}, Z should now be up. The Billboard vector is correct, but when passing the MouseEvent info to the CameraUI, the Camera still wants to rotate as if Y is up.

Thanks!

Bradley

**Comment by paulhoux** [comment]
Ah, OK, thanks for your reply. Will look at it again tonight.

**Comment by bradrothenberg** [comment]
Paul,

Did you get a chance to look at this??

it looks like the issue is in line 124 of CameraUi.cpp :

```c++
		bool invertMotion = ( mInitialCam.getOrientation() * glm::vec3( 0, 1, 0 ) ).y < 0.0f;
```

I think glm::vec3(0,1,0) should be replaced with whatever the camera's z-axis is set to...I'm going to try this now...

Bradley

**Comment by andrewfb** [comment]
Addressed by #995
